tests/Unit/DataStructures/Tensor/Test_TensorData.py

```diff
@@ -0,0 +1,66 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+from spectre.DataStructures import (DataVector, ExtentsAndTensorVolumeData,
+                                    TensorComponent)
+import unittest
+import numpy as np
+import numpy.testing as npt
+
+
+class TestTensorData(unittest.TestCase):
+    # Tests for TensorComponent functions
+    def test_tensor_component(self):
+        # Set up Tensor Component
+        tensor_component = TensorComponent("tensor component",
+                                           DataVector([1.5, 1.1]))
+        # Test name
+        self.assertEqual(tensor_component.name, "tensor component")
+        tensor_component.name = "new tensor component"
+        self.assertEqual(tensor_component.name, "new tensor component")
+        # Test data
+        npt.assert_array_almost_equal(np.array(tensor_component.data),
+                                      np.array([1.5, 1.1]))
```

Comment:
you don't need to compare approximately, you can just use:

```py
self.assertEqual(tensor_component.data, DataVector([1.5, 1.1]))
```

(same below)

It's a small change so if you don't want to bother that's ok, but while you're deleting the one include (other comment) you might as well make this change